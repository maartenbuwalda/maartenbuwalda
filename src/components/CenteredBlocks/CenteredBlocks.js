import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { sizes } from '../../constants/mediaQueries'

const Centered = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 1rem;

  @media (min-width: ${sizes.m}) {
    grid-template-columns: 1fr 1fr;
  }

  @media (min-width: ${sizes.l}) {
    grid-template-columns: 1fr 1fr 1fr;
  }
`

const CenteredBlocks = ({ children }) => {
  return <Centered>{children}</Centered>
}

CenteredBlocks.propTypes = {
  children: PropTypes.node,
}

export default CenteredBlocks
