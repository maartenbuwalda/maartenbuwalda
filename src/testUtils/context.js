const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
