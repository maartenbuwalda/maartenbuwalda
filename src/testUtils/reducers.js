export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
