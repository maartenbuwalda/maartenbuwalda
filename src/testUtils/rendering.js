import toJson from 'enzyme-to-json'
import { shallow, mount } from 'enzyme'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  })
}

export const componentMounts = component => {
  it('mounts correctly', () => {
    const mounted = mount(component)
    expect(toJson(mounted)).toMatchSnapshot()
  })
}
