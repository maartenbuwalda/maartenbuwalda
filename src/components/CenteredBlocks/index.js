import CenteredBlocks from './CenteredBlocks'

export default CenteredBlocks
