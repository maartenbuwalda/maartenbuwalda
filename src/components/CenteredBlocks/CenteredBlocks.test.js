import React from 'react'
import CenteredBlocks from './index'
import { componentRenders } from '../../testUtils/rendering'

describe('CenteredBlocks component', () => {
  const component = <CenteredBlocks>Children</CenteredBlocks>
  componentRenders(component)
})
