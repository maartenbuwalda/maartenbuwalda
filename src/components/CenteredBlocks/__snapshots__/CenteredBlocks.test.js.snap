// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CenteredBlocks component renders correctly 1`] = `
<styled.div>
  Children
</styled.div>
`;
